internal/pkg/api/handleAck.go

```diff
@@ -620,6 +620,9 @@ func (ack *AckT) handleUpgrade(ctx context.Context, zlog zerolog.Logger, agent *
 			dl.FieldUpgradeStatus:    nil,
 			dl.FieldUpgradedAt:       now,
 		}
+		if agent.UpgradeDetails == nil {
```

Comment:
do we need to check nil?
we are upgrading UpgradedAt 2 lines above. so we consider this done

Reply:
when there is `upgrade_details`, the `upgrade_attempts` is cleared when the state is `UPG_WATCHING` (see reasoning [here](https://github.com/elastic/fleet-server/pull/4528#issuecomment-2701686020))
this fix is mainly for horde, where there is no `upgrade_details`